modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/util/ReconnectionUtils.java

```diff
@@ -134,21 +136,33 @@ public static InterceptorChain createReconnectionInterceptorsChain(ExtensionMode
     InterceptorChain.Builder chainBuilder = InterceptorChain.builder();
 
     if (componentModel instanceof ConnectableComponentModel) {
-      if (((ConnectableComponentModel) componentModel).requiresConnection()) {
-        addConnectionInterceptors(chainBuilder, extensionModel, componentModel, connectionSupplier, reflectionCache);
+      if (requiresConnectionInterceptors(extensionModel, (ConnectableComponentModel) componentModel)) {
+        addConnectionInterceptors(chainBuilder, connectionSupplier);
+      }
+
+      if (requiresCursorResetInterceptors((ConnectableComponentModel) componentModel)) {
+        addCursorResetInterceptor(chainBuilder, extensionModel, componentModel, reflectionCache);
       }
     }
 
     return chainBuilder.build();
   }
 
+  private static boolean requiresConnectionInterceptors(ExtensionModel extensionModel, ConnectableComponentModel componentModel) {
+    // Only connectable components that require a connection to be provided beforehand should add the connection interceptors
+    return requiresConnectionProvisioning(extensionModel, componentModel);
+  }
+
+  private static boolean requiresCursorResetInterceptors(ConnectableComponentModel componentModel) {
+    // Connectable components that don't allow for controlling the reconnection strategy should not add the cursor reset
+    // interceptors
+    return componentModel.requiresConnection()
+        && !componentModel.getModelProperty(NoReconnectionStrategyModelProperty.class).isPresent();
```

Comment:
already seen this quite a few times. Find all the places in which the presence of `NoReconnectionStrategyModelProperty` is checked and extract to some utils.

Reply:
Done